### DISCUSSION TITLE
Support for i18n auto redirection for non index routes

### DISCUSSION THREAD

**Discussion by jsphkhan** [vercel/next.js#21187] [Ideas]
**Describe the feature you'd like to request**

Currently, the NextJS server auto redirects to the correct locale either by detecting `Accept-Language` header or the `NEXT_LOCALE` cookie. But this works only for the index route. Kindly add support for non-index or internal routes.

Example next.config configuration:
```bash
module.exports = {
    i18n: {
        localeDetection: true,
        locales: ['en', 'ar'],
        defaultLocale: 'en'
    }
}
```

**Describe the solution you'd like**

Pls, provide a solution to have auto redirects for non-index routes. 

**Describe alternatives you've considered**

Right now, I am using a custom NodeJS express server to redirect based on a sub path or cookie. Path being a priority

**Comment by maxgfr** [comment]
+1
